src/modules/endpoint/sagas/loadCategories.js

```diff
@@ -17,37 +17,44 @@ function * fetchCategoriesMap (city: string, language: string): Saga<CategoriesM
   return categoriesPayload.data
 }
 
-function * loadCategories (city: string, language: string, dataContainer: DataContainer, shouldUpdate: boolean): Saga<FetchMapType> {
+function * loadCategories (
+  city: string,
+  language: string,
+  dataContainer: DataContainer,
+  shouldUpdate: boolean
+): Saga<FetchMapType> {
+  let categories: CategoriesMapModel
+
   // If data was loaded and should not be updated, return
-  if (dataContainer.categoriesAvailable() && !shouldUpdate) {
-    console.debug('Using cached categories')
-    return {}
-  }
+  if (!dataContainer.categoriesAvailable() || shouldUpdate) {
+    console.debug('Fetching categories')
 
-  console.debug('Fetching categories')
+    // TODO: data was loaded but should be incrementally updated. This will be done in NATIVE-3
 
-  // TODO: data was loaded but should be incrementally updated. This will be done in NATIVE-3
+    const categoriesMap: ?CategoriesMapModel = yield call(fetchCategoriesMap, city, language)
 
-  const categoriesMap: ?CategoriesMapModel = yield call(fetchCategoriesMap, city, language)
+    if (!categoriesMap) {
+      throw new Error('Failed to load categories!')
+    }
 
-  if (!categoriesMap) {
-    throw new Error('Failed to load categories!')
-  }
+    yield call(dataContainer.setCategoriesMap, categoriesMap)
 
-  const categories = categoriesMap.toArray()
+    categories = categoriesMap
```

Comment:
You could use `categories` instead of using a separate `categoriesMap` var from the start.